core/src/main/kotlin/org/evomaster/core/database/DbActionGeneBuilder.kt

```diff
@@ -0,0 +1,306 @@
+        } else {
+            StringGene(name = column.name, value = "f", minLength = 0, maxLength = 1)
+        }
+    }
+
+    private fun handleDoubleColumn(column: Column): Gene {
+        // TODO How to discover if the source field is a float/Float field?
+        return if (column.enumValuesAsStrings != null) {
+            EnumGene(name = column.name, values = column.enumValuesAsStrings.map { it.toDouble() })
+        } else {
+            DoubleGene(column.name)
+        }
+    }
+
+    private fun handleSmallIntColumn(column: Column): Gene {
+        return if (column.enumValuesAsStrings != null) {
+            EnumGene(column.name, column.enumValuesAsStrings.map { it.toInt() })
+        } else {
+            IntegerGene(column.name,
+                    min = column.lowerBound ?: Short.MIN_VALUE.toInt(),
+                    max = column.upperBound ?: Short.MAX_VALUE.toInt())
+        }
+    }
+
+    private fun handleTinyIntColumn(column: Column): Gene {
+        return if (column.enumValuesAsStrings != null) {
+            EnumGene(column.name, column.enumValuesAsStrings.map { it.toInt() })
+        } else {
+            IntegerGene(column.name,
+                    min = column.lowerBound ?: Byte.MIN_VALUE.toInt(),
+                    max = column.upperBound ?: Byte.MAX_VALUE.toInt())
+        }
+    }
+
+    private fun handleTextColumn(column: Column): Gene {
+        return if (column.enumValuesAsStrings != null) {
+            EnumGene(name = column.name, values = column.enumValuesAsStrings)
+        } else {
+            if (column.similarToPatterns != null && column.similarToPatterns.isNotEmpty()) {
+                val columnName = column.name
+                val similarToPatterns: List<String> = column.similarToPatterns
+                buildSimilarToRegexGene(columnName, similarToPatterns)
+            } else if (column.likePatterns != null && column.likePatterns.isNotEmpty()) {
+                val columnName = column.name
+                val likePatterns = column.likePatterns
+                buildLikeRegexGene(columnName, likePatterns)
+            } else {
+                StringGene(name = column.name, minLength = 0, maxLength = column.size)
+            }
+        }
+    }
+
+    fun buildLikeRegexGene(columnName: String, likePatterns: List<String>): RegexGene {
+        val disjunctionRxGenes = likePatterns
+                .map { createGeneForPostgresLike(it) }
+                .map { it.disjunctions }
+                .map { it.disjunctions }
+                .flatten()
+        val disjunctionListRxGene = DisjunctionListRxGene(disjunctions = disjunctionRxGenes)
+        return RegexGene(columnName, disjunctions = disjunctionListRxGene)
+    }
+
+    fun buildSimilarToRegexGene(columnName: String, similarToPatterns: List<String>): RegexGene {
+        val joinedPattern = similarToPatterns.map { c -> "(%s)".format(c) }.joinToString("|")
```

Comment:
i do not understand this `"(%s)".format(c)`. can u add a comment?

Reply:
I changed the implementation of buildSimilarToRegexGene() to match buildLikeRegexGene()